modules/standard/Map.chpl

```diff
@@ -737,8 +737,15 @@ module Map {
      left-hand map, but not the right-hand map. */
   proc -=(ref a: map(?keyType, ?valueType, ?parSafe),
           b: map(keyType, valueType, parSafe)) {
-    for k in b.keys() do
-      a.remove(k);
+    a._enter(); defer a._leave();
+    for k in b.keys() {
+      var (found, slot) = a.table.findFullSlot(k);
+      if found {
+        var outKey: keyType, outVal: valueType;
+        a.table.clearSlot(slot, outKey, outVal);
+      }
+    }
+    a.table.maybeShrinkAfterRemove();
```

Comment:
By my read, ChapelHashtable's `maybeShrinkAfterRemove()` will shrink the hashtable by at most one step down in the `chpl__primes` table, even if this `-=` removed all the elements in a very large map.  Is that ok?